fuzz/fuzz_targets/c_rust_merkle.rs

```diff
@@ -55,8 +55,8 @@ fn main() {
     }
 }
 
-#[cfg(test)]
-mod unit_tests {
+#[cfg(all(test, fuzzing))]
```

Comment:
In 2a3b18e919b54cea75383ae03ac2b83f0e002d88:

Why gate on `fuzzing`? This is super annoying.

Reply:
[I did the same as in rust-bitcoin](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/fuzz/fuzz_targets/bitcoin/deser_net_msg.rs#L16).

Reply:
If anything is broken by `cfg(fuzzing)` then we should fix the broken stuff, not inconvenience every user of the fuzztests ever.